### PR TITLE
235: Filter NodeJS dev dependencies

### PR DESCRIPTION
**Pull request by bsima**
This should be good to go now.

---

Closes https://github.com/fossas/fossa-cli/issues/235

### CODE REVIEW

File: buildtools/npm/npm_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+    "testing"
+
+    "github.com/fossas/fossa-cli/buildtools/npm"
+)
+
+func TestNoProdDeps(t *testing.T) {
+    dir := "buildtools/npm/testdata"
+    n, e := npm.New()
```
[the remark this answers is not in the document]
Comment (bsima) [reply buildtools/npm/npm_test.go]: ok, will fix

**Comment by bsima** [comment]
Okay, I'll add mocks

File: analyzers/nodejs/nodejs_test.go [hunk 2]
```diff
@@ -131,3 +131,22 @@ func TestUsingNodeModuleFallback(t *testing.T) {
...
+    m := module.Module{
+		Dir:  filepath.Join("testdata", "prodonly"),
+		Type: pkg.NodeJS,
+	}
+
+	a, err := analyzers.New(m)
+	assert.NoError(t, err)
+
+	a.(*nodejs.Analyzer).Tool = MockNPM{
+		JSONFilename: filepath.Join("testdata", "prodonly", "package.json"),
```
Comment (liftM) [inline analyzers/nodejs/nodejs_test.go]: This should actually be the route to a fixture with the output of `npm ls --json`, not to a manifest directly.

**Comment by bsima** [comment]
I tested this manually with the example package.json I added and it works.

**Comment by bsima** [comment]
Not sure what this circleci error means though:

```
#!/bin/bash -eo pipefail
make -s junit-test > $TEST_RESULTS/go-test-report.xml

Bad response status from coveralls: 422
{"message":"Couldn't find a repository matching this job.","error":true}
make: *** [junit-test] Error 1
Exited with code 2
```

**Comment by liftM** [comment]
Closed by #314.

**Pull request closed without merging**